src/components/board/Board.jsx

```diff
@@ -6,42 +6,64 @@ import { useCallback, useEffect, useRef } from 'react';
 import { useSelector } from 'react-redux';
 import { useParams } from 'react-router';
 import { useDraggable } from 'react-use-draggable-scroll';
-import { loadBoard } from '../../store/board/board.actions.js';
+import { loadBoard, updateBoard } from '../../store/board/board.actions.js';
 import { BoardHeader } from './components/BoardHeader.jsx';
 import { Column } from './components/Column.jsx';
 
 export function Board() {
   const { boardId } = useParams();
   const board = useSelector((state) => state.boardModule.board);
+  console.log('Board data:', board);
   const listRef = useRef();
   const { event } = useDraggable(listRef);
+  
+  useEffect(() => {
+    loadBoard(boardId);
+  }, [boardId]);
+
 
-  const moveTask = useCallback(
-    ({
-      taskIndexInStartGroup,
-      sourceGroupId,
-      destinationGroupId,
-      itemIndexInFinishGroup,
-    }) => {
-      const startGroupData = board.groups[sourceGroupId];
-      const destinationGroupData = board.groups[destinationGroupId];
-      const taskToMove = startGroupData.tasksIds[taskIndexInStartGroup];
-      const newStartGroupData = {
-        ...startGroupData,
-        tasksIds: startGroupData.tasksIds.filter((task) => task !== taskToMove),
-      };
-      const newDestinationTasks = destinationGroupData.tasksIds;
-      const newIndexInDestination = itemIndexInFinishGroup ?? 0;
-
-      newDestinationTasks.splice(newIndexInDestination, 0, taskToMove);
-
-      const newFinishGroupData = {
-        ...destinationGroupData,
-        tasksIds: newDestinationTasks,
-      };
-
-      console.log('moveTask newStartGroupData ', newStartGroupData);
-      console.log('moveTask newFinishGroupData ', newFinishGroupData);
+ const moveTask = useCallback(
+   ({
+     taskIndexInStartGroup,
+     sourceGroupId,
+     destinationGroupId,
+     itemIndexInFinishGroup,
+   }) => {
+     if (!board) return;
```

Comment:
why you need here board condition?

Reply:
We don't have to, but i used to do it every where i am afraid to get null/undefined. but acutally we can make it only where we fetch board and not inside the group/task... i will remove it if u dont want...